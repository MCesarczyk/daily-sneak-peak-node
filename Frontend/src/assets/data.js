export const AUTHOR_PAGE_ADDRESS = "https://home.cesarczyk.dev";
export const AUTHOR_PAGE_LABEL = "Cesarczyk.dev";
export const DEMO_DELAY = 300;