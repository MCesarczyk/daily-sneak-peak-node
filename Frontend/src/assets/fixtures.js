export const pages = [
  {
    id: 1,
    label: 'Children',
    path: '/children'
  },
  {
    id: 2,
    label: 'Groups',
    path: '/groups'
  },
  {
    id: 3,
    label: 'Users',
    path: '/users'
  }
];

export const modal = {
  addChild: {
    buttonText: "Add+",
    title: "Add new child"
  },
  editChild: {
    buttonText: "Edit",
    title: "Edit child"
  },
  addGroup: {
    buttonText: "Add+",
    title: "Add new group"
  },
  editGroup: {
    buttonText: "Edit",
    title: "Edit group"
  },
  addActivities: {
    buttonText: "Add+",
    title: "Add daily activities"
  },
  editActivities: {
    buttonText: "Edit",
    title: "Edit activities"
  }
}