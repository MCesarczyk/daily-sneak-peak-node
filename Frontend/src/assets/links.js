const apiUrl = process.env.NODE_ENV === 'production' ? 'http://164.92.135.59:5000' : '';

export const TITLE_URL = `${apiUrl}/api/title`;
export const CHILDREN_LIST_URL = `${apiUrl}/api/get/children`;
export const CHILD_POST_URL = `${apiUrl}/api/post/childtodb`;
export const CHILD_UPDATE_URL = `${apiUrl}/api/put/children`;
export const CHILD_DELETE_URL = `${apiUrl}/api/delete/children`;