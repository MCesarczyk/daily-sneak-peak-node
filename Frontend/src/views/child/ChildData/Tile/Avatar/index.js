import { useFetchData } from "../../../../../assets/utils/useFetchData";
import { ImageOuterWrapper, ImageWrapper, StyledImage } from "./styled";
import { NoPicture } from "./NoPicture";

const Avatar = ({ url }) => {
  const picture = useFetchData(url);

  return (
    <ImageOuterWrapper>
      <ImageWrapper>
        {(url === '' || url === undefined) ? <NoPicture /> : <StyledImage src={picture} />}
      </ImageWrapper>
    </ImageOuterWrapper>
  )
};

export default Avatar;