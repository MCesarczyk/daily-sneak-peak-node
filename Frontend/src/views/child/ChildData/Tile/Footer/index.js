import React from "react";
import { Divider } from "@mui/material";
import { ButtonsWrapper, StyledFooter } from "./styled";
import DialogPopup from "../../../../dialog/DialogPopup";
import Confirmation from "../../../../../components/Confirmation";

const TileFooter = ({ data }) => (
  <StyledFooter>
    <Divider />
    <ButtonsWrapper>
      <Confirmation
        buttonLabel={data?.buttonLabel}
        popupContent={data?.onDeleteWarning}
        onActionCall={data?.onActionCall}
      />
      <DialogPopup
        form={data?.popupFormType}
      />
    </ButtonsWrapper>
  </StyledFooter>
);

export default TileFooter;