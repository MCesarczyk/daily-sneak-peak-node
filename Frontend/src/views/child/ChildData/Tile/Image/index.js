import { NoPicture } from "./NoPicture";
import { ImageOuterWrapper, ImageWrapper } from "./styled";
import Portrait from "./Portrait";

const Image = ({ child }) => (
  <ImageOuterWrapper>
    <ImageWrapper>
      {(child?.avatarUrl === '' || child?.avatarUrl === undefined) ? <NoPicture /> : <Portrait />}
    </ImageWrapper>
  </ImageOuterWrapper>
);

export default Image;