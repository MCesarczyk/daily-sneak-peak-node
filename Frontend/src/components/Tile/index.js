import React from "react";
import { Skeleton } from "@mui/material";
import TileFooter from "./Footer";
import NotFound from "../NotFound";
import { StyledTile } from "./styled";
import Avatar from "./Avatar";
import Heading from "./Heading";
import Loader from "../Loader";

const Tile = ({ state, componentsState, avatarUrl, headingData, footerData }) => (
  <Loader
    state={state}
    message="Loading"
  >
    {componentsState.notFound
      ? <NotFound message={componentsState.errorMessage} />
      : <StyledTile data-testid="details-tile">
        <Avatar url={avatarUrl} />
        {componentsState.loading ?
          <>
            <Skeleton height={42} />
            <Skeleton height={36} />
          </>
          :
          <Heading data={headingData} />
        }
        <TileFooter data={footerData} />
      </StyledTile>
    }
  </Loader>
);

export default Tile;