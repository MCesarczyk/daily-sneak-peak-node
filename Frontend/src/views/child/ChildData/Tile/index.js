import React from "react";
import { Skeleton } from "@mui/material";
import TileFooter from "./Footer";
import NotFound from "../../../../components/NotFound";
import { StyledTile } from "./styled";
import Image from "./Image";
import Heading from "./Heading";

const Tile = ({ child, headingData, footerData }) => (
  <>
    {!child
      ? <NotFound message="Sorry... no child data found." />
      : <StyledTile data-testid="child-tile">
        <Image child={child} />
        {child && Object.entries(child).length === 0 ?
          <>
            <Skeleton height={42} />
            <Skeleton height={36} />
          </>
          :
          <Heading data={headingData} />
        }
        <TileFooter data={footerData} />
      </StyledTile>
    }
  </>
);

export default Tile;