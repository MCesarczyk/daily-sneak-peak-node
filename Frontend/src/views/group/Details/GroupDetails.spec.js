import reducer from "../groupSlice";
import mock from "../../../assets/mocks/group.json";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../../store";
import { ThemeProvider } from "styled-components";
import { lightMode } from "../../../assets/theme";
import Tile from "../../../components/Tile";

const setup = (state, group) => render(
  <Provider store={store}>
    <ThemeProvider theme={lightMode}>
      <Tile
        state={state}
        componentsState={{
          notFound: !group,
          errorMessage: "Sorry... no group data found.",
          loading: group && Object.entries(group).length === 0
        }}
        avatarUrl={group?.avatarUrl}
        headingData={{
          title: group?.name,
          subtitle: group?.description,
          titleLabel: "group name: ",
          subtitleLabel: "description: "
        }}
        footerData={{
          buttonLabel: "Delete group",
          onDeleteWarning: `You're about to delete group: ${group?.name}`,
          onActionCall: () => dispatch(deleteGroupData(group?.id)),
          popupFormType: 'editGroup'
        }}
      />
    </ThemeProvider>
  </Provider>
);

test("Tile component should render spinner during loading", () => {
  setup('loading');

  const loader = screen.queryByText(/Loading/);
  expect(loader).toBeInTheDocument();
});

test("Groups list should be empty in idle state", () => {
  const { container } = setup('idle');

  expect(container.firstChild).toBeNull();
});

test("Group reducer should return the initial state", () => {
  expect(reducer(undefined, {})).toEqual(
    {
      id: 0,
      data: {},
      state: 'idle',
      gotoList: false
    }
  )
});

test("Group reducer should return mocked data", () => {
  const previousState = mock;
  expect(reducer(previousState, {})).toMatchSnapshot();
});

test("Group data should render properly", async () => {
  setup('success', mock);

  const tile = screen.queryByTestId('details-tile');
  expect(tile).toBeInTheDocument();

  expect(screen.getByText(/group name:/)).toBeInTheDocument();
  expect(screen.getByRole('heading').textContent).toBe('Crocodiles');
  expect(screen.getByText(/description:/)).toBeInTheDocument();
  expect(screen.getByRole('subheading').innerHTML).toBe("6year olders");
});

test("Placeholder should display when it's no data", () => {
  setup('success');

  const tile = screen.queryByTestId('details-tile');
  expect(tile).not.toBeInTheDocument();

  const errorMessage = screen.getByRole('heading');
  expect(errorMessage.innerHTML).toBe('Sorry... no group data found.');

  const errorImage = screen.queryByRole('img');
  expect(errorImage).toBeInTheDocument();
});