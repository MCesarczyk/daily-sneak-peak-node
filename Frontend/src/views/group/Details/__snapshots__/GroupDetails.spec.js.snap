// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group reducer should return mocked data 1`] = `
Object {
  "avatar": "",
  "created_at": "2022-04-13T16:43:19.248Z",
  "description": "6year olders",
  "gid": 3,
  "name": "Crocodiles",
  "notes": "some interesting details",
  "updated_at": "2022-04-13T16:43:19.248Z",
}
`;
